db-engine/src/main/java/com/splicemachine/db/catalog/SystemProcedures.java

```diff
@@ -2080,6 +2080,31 @@ public static void SYSCS_SET_USER_ACCESS(String userName, String connectionPermi
     }
 
     /**
+     * Save (or delete) the source text for a user-defined code object in the SYS.SYSSOURCECODE table
+     *
+     * @param schemaName name of the user that owns the code object
+     * @param objectName name of the code object
+     * @param objectType type of the code object (e.g. "Function")
+     * @param objectForm format of the source text (e.g. "Java8@UTF8")
+     * @param definerName name of the user saving the object
+     * @param sourceCode source text of the code object (or null to request deletion of the code object)
+     * @throws SQLException Error saving the sourcecode
+     */
+    public static void SYSCS_SAVE_SOURCECODE(String schemaName, String objectName, String objectType, String objectForm, String definerName, Blob sourceCode) throws SQLException{
+        LanguageConnectionContext lcc = ConnectionUtil.getCurrentLCC();
+        TransactionController tc = lcc.getTransactionExecute();
+        try{
+            tc.elevate("sourceCode");
+            DataDictionary dd = lcc.getDataDictionary();
+            SourceCodeDescriptor descriptor = new SourceCodeDescriptor(schemaName, objectName, objectType, objectForm, definerName, sourceCode);
+            dd.saveSourceCode(descriptor, tc);
+
+        }catch(StandardException se){
+            throw PublicAPI.wrapStandardException(se);
+        }
+    }
+
+    /**
      * Utility method for SYSCS_SET_USER_ACCESS removes a user from
```

Comment:
Move this to SpliceAdmin.java. This is where we put our system procedure implementation.